# Carga de datos

**Proyecto Integrador — Modelo de riesgo crediticio**

Este notebook no se utilizaría normalmente en un proceso productivo de creación
de modelos, dado que la información debería estar disponible en el DWH /
Datalake de la empresa como resultado de un proceso de ingesta distinto. Sin
embargo, para este ejercicio académico se utiliza un dataset **no productivo**
de ejemplo en formato `.csv` (`Base_de_datos.csv`), ubicado en la raíz del
proyecto.

Objetivo de este notebook: cargar el dataset y realizar una validación
estructural básica (dimensiones, tipos de datos, duplicados, primeras filas)
antes de pasar al análisis exploratorio en `comprension_eda.ipynb`.

In [1]:
import pandas as pd
from pathlib import Path

pd.set_option("display.max_columns", None)

DATA_PATH = Path("..") / "Base_de_datos.csv"
DATA_PATH

PosixPath('../Base_de_datos.csv')

## 1. Carga del dataset

In [2]:
df = pd.read_csv(DATA_PATH)
print(f"Filas: {df.shape[0]:,} | Columnas: {df.shape[1]}")
df.head()

Filas: 10,763 | Columnas: 23


,tipo_credito,fecha_prestamo,capital_prestado,plazo_meses,edad_cliente,tipo_laboral,salario_cliente,total_otros_prestamos,cuota_pactada,puntaje,puntaje_datacredito,cant_creditosvigentes,huella_consulta,saldo_mora,saldo_total,saldo_principal,saldo_mora_codeudor,creditos_sectorFinanciero,creditos_sectorCooperativo,creditos_sectorReal,promedio_ingresos_datacredito,tendencia_ingresos,Pago_atiempo
0,7,2024-12-21 11:31:35,3692160.0,10,42,Independiente,8000000,2500000,341296,88.768094,695.0,10,5,0.0,51258.0,51258.0,0.0,5,0,0,908526.0,Estable,1
1,4,2025-04-22 09:47:35,840000.0,6,60,Empleado,3000000,2000000,124876,95.227787,789.0,3,1,0.0,8673.0,8673.0,0.0,0,0,2,939017.0,Creciente,1
2,9,2026-01-08 12:22:40,5974028.4,10,36,Independiente,4036000,829000,529554,47.613894,740.0,4,5,0.0,18702.0,18702.0,0.0,3,0,0,NaN,NaN,0
3,4,2025-08-04 12:04:10,1671240.0,6,48,Empleado,1524547,498000,252420,95.227787,837.0,4,4,0.0,15782.0,15782.0,0.0,3,0,0,1536193.0,Creciente,1
4,9,2025-04-26 11:24:26,2781636.0,11,44,Empleado,5000000,4000000,217037,95.227787,771.0,4,6,0.0,204804.0,204804.0,0.0,3,0,1,933473.0,Creciente,1


## 2. Validación estructural básica

In [3]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 10763 entries, 0 to 10762
Data columns (total 23 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   tipo_credito                   10763 non-null  int64  
 1   fecha_prestamo                 10763 non-null  str    
 2   capital_prestado               10763 non-null  float64
 3   plazo_meses                    10763 non-null  int64  
 4   edad_cliente                   10763 non-null  int64  
 5   tipo_laboral                   10763 non-null  str    
 6   salario_cliente                10763 non-null  int64  
 7   total_otros_prestamos          10763 non-null  int64  
 8   cuota_pactada                  10763 non-null  int64  
 9   puntaje                        10763 non-null  float64
 10  puntaje_datacredito            10757 non-null  float64
 11  cant_creditosvigentes          10763 non-null  int64  
 12  huella_consulta                10763 non-null  int64  
 1

In [4]:
# Filas duplicadas
n_dup = df.duplicated().sum()
print(f"Filas completamente duplicadas: {n_dup}")

Filas completamente duplicadas: 0


In [5]:
# Nulos por columna (revisión preliminar; el tratamiento detallado se hace en comprension_eda.ipynb)
df.isnull().sum().to_frame("nulos")

,nulos
tipo_credito,0
fecha_prestamo,0
capital_prestado,0
plazo_meses,0
edad_cliente,0
tipo_laboral,0
salario_cliente,0
total_otros_prestamos,0
cuota_pactada,0
puntaje,0


## 3. Resultado

El dataset se cargó correctamente: **10,763 filas** y **23 columnas**, sin
filas duplicadas. Se identifican nulos en varias columnas (`puntaje_datacredito`,
`saldo_mora`, `saldo_total`, `saldo_principal`, `saldo_mora_codeudor`,
`promedio_ingresos_datacredito`, `tendencia_ingresos`), los cuales se analizan
y tratan en detalle en `comprension_eda.ipynb`.

La variable objetivo del proyecto es **`Pago_atiempo`** (1 = pagó a tiempo,
0 = no pagó a tiempo).